### PR TITLE
Gp 2121

### PR DESCRIPTION
**Pull request by pruzanov**
Initial import of tested wdl workflow

### CODE REVIEW

**Comment by pruzanov** [comment]
retest this please

File: hsmetrics.wdl [hunk 1]
```diff
@@ -0,0 +1,160 @@
+version 1.0
+
+workflow hsMetrics {
+input {
+    File    inputBam
+    File    baitBed
+    File    targetBed
+    String  refFasta
+    String? outputPrefix = ""
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: `outputPrefix` -> `outputFileNamePrefix`

File: hsmetrics.wdl [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+# ==========================================
+#  TASK 1 of 4: make reference dictionary
+# ==========================================
+task makeRefDictionary {
+input {
+        String refFasta
+        Int?   jobMemory  = 18
+        Int?   javaMemory = 10
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: Can this be modified to one memory parameter and then something like `~{jobMemory -4}` in the command block

File: hsmetrics.wdl [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  File outputINSMetrics = collectInsertMetrics.outputINSMetrics
+  File outputINSPDF     = collectInsertMetrics.outputINSPDF
+}
+
+}
+
+# ==========================================
+#  TASK 1 of 4: make reference dictionary
+# ==========================================
+task makeRefDictionary {
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: This should be moved to `modulator`

File: hsmetrics.wdl [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+# ==========================================
+#  TASK 2 of 4: convert bed to intervals
+# ==========================================
+task bedToIntervals {
+input {
+	File   inputBed
+        File   refDict       
+        Int?   jobMemory  = 18
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: One memory parameter and ~{jobMemory -4} in the command block

File: hsmetrics.wdl [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+output {
+  File refDict = "~{basename(refFasta, '.fa')}.dict"
+}
+}
+
+# ==========================================
+#  TASK 2 of 4: convert bed to intervals
+# ==========================================
+task bedToIntervals {
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: This data should be moved to modulator

File: hsmetrics.wdl [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+input { 
+        File   inputBam
+        File   baitIntervals
+        File   targetIntervals
+        String refFasta
+        String? metricTag  = "HS"
+        String? filter     = "LENIENT"
+        String? outputPrefix = ""
+        Int?   jobMemory   = 18
+        Int?   javaMemory  = 12
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: One memory parameter and ~{jobMemory -4} in the command block

File: hsmetrics.wdl [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+
+command <<<
+ java -Xmx~{javaMemory}G -jar $PICARD_ROOT/picard.jar CollectHsMetrics \
+                              TMP_DIR=picardTmp \
+                              BAIT_INTERVALS=~{baitIntervals} \
+                              TARGET_INTERVALS=~{targetIntervals} \
+                              R=~{refFasta} \
+                              COVERAGE_CAP=~{coverageCap} \
+                              INPUT=~{inputBam} \
+                              OUTPUT="~{basename(inputBam, '.bam')}~{outputPrefix}.~{metricTag}" \
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: `outputPrefix` should be ~{basename(inputBam, '.bam')}

File: hsmetrics.wdl [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                              VALIDATION_STRINGENCY=~{filter} 
+>>>
+
+runtime {
+  memory:  "~{jobMemory} GB"
+  modules: "~{modules}"
+}
+
+output {
+  File outputHSMetrics = "~{basename(inputBam, '.bam')}~{outputPrefix}.~{metricTag}"
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: Just `~{outputPrefix}`

File: hsmetrics.wdl [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+# ==========================================
+#  TASK 4 of 4: collect Insert metrics
+# ==========================================
+task collectInsertMetrics {
+input {
+        File    inputBam
+        String? metricTag  = "INS"
+        Int?    jobMemory  = 18
+        Int?    javaMemory = 12
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: One memory parameter and ~{jobMemory -4} in the command block

File: hsmetrics.wdl [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        Float?  minPct     = 0.5
+        String? outputPrefix = ""
+        String? modules    = "java/8 rstats/3.6 picard/2.19.2"
+}
+
+command <<<
+ java -Xmx~{javaMemory}G -jar $PICARD_ROOT/picard.jar CollectInsertSizeMetrics \
+                              INPUT=~{inputBam} \
+                              OUTPUT="~{basename(inputBam, '.bam')}~{outputPrefix}.~{metricTag}" \
+                              H="~{basename(inputBam, '.bam')}~{outputPrefix}.~{metricTag}.PDF" \
```
Comment (mlaszloffy) [inline hsmetrics.wdl]: outputPrefix default = `~{basename(inputBam, '.bam')}`

**Review by mlaszloffy — CHANGES_REQUESTED** [review]

File: hsmetrics.wdl (hunk 1 above)
Comment (pruzanov) [reply hsmetrics.wdl]: fixed

File: hsmetrics.wdl (hunk 2 above)
Comment (pruzanov) [reply hsmetrics.wdl]: implemented

File: hsmetrics.wdl (hunk 4 above)
Comment (pruzanov) [reply hsmetrics.wdl]: implemented

File: hsmetrics.wdl (hunk 6 above)
Comment (pruzanov) [reply hsmetrics.wdl]: fixed

File: hsmetrics.wdl (hunk 7 above)
Comment (pruzanov) [reply hsmetrics.wdl]: provided a default

File: hsmetrics.wdl (hunk 8 above)
Comment (pruzanov) [reply hsmetrics.wdl]: fixed

File: hsmetrics.wdl (hunk 9 above)
Comment (pruzanov) [reply hsmetrics.wdl]: fixed

File: hsmetrics.wdl (hunk 10 above)
Comment (pruzanov) [reply hsmetrics.wdl]: fixed

File: hsmetrics.wdl (hunk 3 above)
Comment (pruzanov) [reply hsmetrics.wdl]: Version 1.1 ?

File: hsmetrics.wdl (hunk 5 above)
Comment (pruzanov) [reply hsmetrics.wdl]: this is a very light task, and if we want to have these in modulator we will probably want to have all .bed files in modulator as well. Maybe Version 1.1?

**Comment by pruzanov** [comment]
retest this please

File: hsmetrics.wdl (hunk 3 above)
Comment (pruzanov) [reply hsmetrics.wdl]: Actually, I have this fixed in a good way - modules for hg19 and hg38 with .dict files are currently awaiting Bob's approval and merge

File: hsmetrics.wdl (hunk 5 above)
Comment (mlaszloffy) [reply hsmetrics.wdl]: Sounds good.

**Review by mlaszloffy — APPROVED** [review]

**Review by prisnir-zz — APPROVED** [review]

**Pull request merged**